test/dlc_tests/sanity/quick_checks/test_formatting_and_linting.py

```diff
@@ -0,0 +1,27 @@
+import os
+import pytest
+
+from invoke.context import Context
+
+from test.test_utils import is_pr_context
+
+
+@pytest.mark.quick_checks
+@pytest.mark.model("N/A")
+@pytest.mark.integration("python_style_formatting")
+@pytest.mark.skipif(
+    not is_pr_context(),
+    reason="This test is only needed to validate formatting in PRs.",
+)
+def test_black_formatting():
+    """
+    Check that black style formatting exits gracefully.
+    """
+    # Look up the path until deep-learning-containers is in our base directory
+    dlc_base_dir = os.getcwd()
+    while "deep-learning-containers" not in os.path.basename(dlc_base_dir):
+        dlc_base_dir = os.path.split(dlc_base_dir)[0]
+
+    ctx = Context()
+
+    ctx.run(f"black --check {dlc_base_dir}", hide=True)
```

Comment:
Iirc `black` logs errors to stderr. It might be worthwhile to include this output in our test logging by changing the `hide` kwarg value. Reason being that different versions of `black` can sometimes have differences in formatting requirements.

I have personal experience where `black` failed in pipelines and passed locally for this exact reason. It can be very confusing and having any sort of error output from the tool would provide some context for a formatting check failure.

Reply:
hide=True is necessary to keep output with pytest stacktrace when running tests in parallel. It is an edge case with invoke/pytest behavior, you'll see that throughout our codebase.

It is not as valuable, imo to have the output print somewhere high up in the logs as opposed to along with the stack trace. So as implemented, I think it it will be easier to work with.